### PR TITLE
Add HA configuration

### PR DESCRIPTION
**Pull request by skonto**
Adds HA options. Relates to https://github.com/mesosphere/dcos-flink-service/pull/23

### CODE REVIEW

**Comment by joerg84** [comment]
LGTM,
a) I still need to test this.
b) @ryadav88 this only adds new config options, would you still prefer a new package version for this?

**Comment by ryadav88** [comment]
@skonto @joerg84 yes, please update the package version and also bump the subdirectory version to 2.

**Comment by skonto** [comment]
@ryadav88 I thought so I will update it accordingly, thnx.

**Comment by skonto** [comment]
@ryadav88 @joerg84  I moved it to a new subdir, current version of the package is `1.2.0-1.4`. If I update that then docker image names need update as well since it depends on it. Should I update everything and then you push both images and code? Let me know how to proceed pls.

**Comment by ryadav88** [comment]
@skonto Yes, I would suggest updating the image too. Also, the default install of the package doesnt work. Please review the dcos example here and create a PR to reflect any changes https://github.com/dcos/examples/tree/master/flink

**Comment by skonto** [comment]
@ryadav88 what do you mean by 

>  Also, the default install of the package doesnt work.

I have tested with a flink could you point me to issues or logs?

**Comment by skonto** [comment]
WIP

**Comment by skonto** [comment]
@joerg84 @ryadav88 | will put HA in 3 version. Sorry for the delay updating this.

**Comment by OhRobin** [comment]
Default installation worked for me, but launching a job via the CLI does not work. Seems to be an issue with the jobmanager host being `localhost`

File: repo/packages/F/flink/2/config.json [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        },
+        "memory-preallocation": {
+          "description": "Specify whether TaskManager-memory should be allocated when starting up (checked) or when memory is required in the memory manager (unchecked).",
+          "type": "boolean",
+          "default": true
+        }
+      }
+    },
+
+    "high-availability": {
```
Comment (EronWright) [inline repo/packages/F/flink/2/config.json]: If possible, the defaults should support a side-by-side installation of numerous instances of the package.    We should use the service name as a component of the various ZK/HDFS paths and as the `cluster-id`.  I'm worried that users will get into serious trouble, with multiple Flink clusters interfering with each other.

File: repo/packages/F/flink/2/config.json [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+      "description":"Flink high availability options.",
+      "type":"object",
+      "properties": {
+        "enabled": {
+          "default":true,
```
Comment (EronWright) [inline repo/packages/F/flink/2/config.json]: I agree that HA should be enabled by default, iff we fix the side-by-side install issue.

**Review by EronWright — CHANGES_REQUESTED** [review]

**Comment by skonto** [comment]
@OhRobin you need to use the jobmanager's correct ip... Is the comment referring to this PR?

File: repo/packages/F/flink/2/config.json (hunk 1 above)
Comment (skonto) [inline repo/packages/F/flink/2/config.json]: This is how people install different versions of the same package. They use a different service name. Ok will fix/check config options to comply with that.

File: repo/packages/F/flink/2/config.json (hunk 2 above)
Comment (skonto) [inline repo/packages/F/flink/2/config.json]: :+1:

**Comment by skonto** [comment]
@EronWright @joerg84  I updated the PR. Will test and report back.

**Comment by skonto** [comment]
Here is the guide how to run an example with the default settings:
https://gist.github.com/skonto/c8404d43142070dca860dda431459887
I will add an example to the example repos.

**Comment by skonto** [comment]
https://github.com/dcos/examples/pull/196

**Comment by joerg84** [comment]
@skonto Thx, will check tomorrow!

**Comment by joerg84** [comment]
@skonto I will incorporate these changes in my universe PR for the Flink  release, ok?

**Comment by skonto** [comment]
@joerg84 sure.

**Comment by skonto** [comment]
@joerg84 should I update the PR?

**Comment by skonto** [comment]
@ryadav88 what should I do with the PR? I am not sure if @joerg84 has merged with his work. Guys?

**Comment by ryadav88** [comment]
@joerg84 any update here?

**Comment by ryadav88** [comment]
Package updates have been merged since this. Closing the PR.

**Comment by skonto** [comment]
@ryadav88 I dont see HA merged somewhere? Why close this?

**Comment by ryadav88** [comment]
@skonto I can't remember now why I closed it then. @joerg84-  do you previous changes cover this? If not, we should reopen this PR. Let me know.

**Comment by skonto** [comment]
@ryadav88 @joerg84 ?

**Pull request closed without merging**